### PR TITLE
BD-18 feat: add forgot password and logic

### PR DESCRIPTION
**Pull request by pRomKol**

### CODE REVIEW

File: components/sign-in-form/SignInForm.tsx [hunk 1]
```diff
@@ -1,14 +1,15 @@
...
 import { zodResolver } from '@hookform/resolvers/zod'
 import Link from 'next/link'
 import { useRouter } from 'next/router'
 import { Button, Input, PasswordInput, Typography } from 'uikit-inctagram'
 import { z } from 'zod'
 
+import { authActions } from '../../features'
+import { authApi, isApiError, isFetchBaseQueryError } from '../../services/'
+import { useAppDispatch } from '../../store'
+
```
Comment (sfgardev) [inline components/sign-in-form/SignInForm.tsx:12]: where did u get this imports, looks like old commit

File: components/sign-in-form/SignInForm.tsx [hunk 2]
```diff
@@ -17,7 +18,7 @@ const SignInFormSchema = z.object({
 type FormFields = z.infer<typeof SignInFormSchema>
 
 export const SignInForm = () => {
-  const [login, { isLoading }] = authApi.useLoginMutation()
+  const [login, { error, isError, isLoading }] = authApi.useLoginMutation()
```
Comment (sfgardev) [inline components/sign-in-form/SignInForm.tsx:21]: looks like old commit

File: components/sign-in-form/SignInForm.tsx [hunk 3]
```diff
@@ -34,10 +35,7 @@ export const SignInForm = () => {
       const response = await login(values).unwrap()
 
       dispatch(authActions.login({ accessToken: response.accessToken }))
-      const payload = response.accessToken.split('.')[1]
-      const id = JSON.parse(atob(payload)).userId
-
-      router.push(`/profile/${id}`)
+      router.push('/profile')
```
Comment (sfgardev) [inline components/sign-in-form/SignInForm.tsx:38]: old commit

File: package.json [hunk 4]
```diff
@@ -16,15 +16,17 @@
   "dependencies": {
     "@hookform/resolvers": "3.9.0",
     "@reduxjs/toolkit": "2.2.7",
+    "@types/react-google-recaptcha": "^2.1.9",
```
Comment (sfgardev) [inline package.json:19]: should be in dev dependencies and version should be fixed by deleting ^

File: package.json [hunk 5]
```diff
@@ -16,15 +16,17 @@
...
     "async-mutex": "0.5.0",
     "next": "14.2.6",
     "next-redux-wrapper": "8.1.0",
     "react": "18",
     "react-dom": "18",
+    "react-google-recaptcha": "^3.1.0",
```
Comment (sfgardev) [inline package.json:25]: version should be fixed

File: services/auth/authSlice.ts [hunk 6]
```diff
@@ -29,6 +30,13 @@ export const authApi = createApi({
         url: 'auth/logout',
       }),
     }),
+    passwordRecovery: builder.mutation<PasswordRecoveryRequest, void>({
```
Comment (sfgardev) [inline services/auth/authSlice.ts:33]: first argument of generic should be response

**Review by sfgardev — CHANGES_REQUESTED** [review]

File: components/forgot-password-form/ForgotPasswordForm.tsx [hunk 7]
```diff
@@ -0,0 +1,113 @@
...
+  }
+
+  const onSubmit: SubmitHandler<FormFields> = async values => {
+    try {
+      const baseUrl = window.location.origin
+
+      await forgotPassword({ baseUrl, ...values }).unwrap()
+      setIsModalOpened(true)
+    } catch (err) {
+      // if (isFetchBaseQueryError(err)) {
```
[the remark this answers is not in the document]
Comment (pRomKol) [reply components/forgot-password-form/ForgotPasswordForm.tsx:45]: done

File: components/sign-in-form/SignInForm.tsx (hunk 1 above)
Comment (pRomKol) [reply components/sign-in-form/SignInForm.tsx:12]: done

File: components/sign-in-form/SignInForm.tsx (hunk 2 above)
Comment (pRomKol) [reply components/sign-in-form/SignInForm.tsx:21]: done

File: components/sign-in-form/SignInForm.tsx (hunk 3 above)
Comment (pRomKol) [reply components/sign-in-form/SignInForm.tsx:38]: done

File: package.json (hunk 4 above)
Comment (pRomKol) [reply package.json]: done

File: package.json (hunk 5 above)
Comment (pRomKol) [reply package.json]: done

File: services/auth/authSlice.ts (hunk 6 above)
Comment (pRomKol) [reply services/auth/authSlice.ts]: done

**Review by sfgardev — APPROVED** [review]

**Pull request merged**